projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -33,7 +33,7 @@
 function register_block() {
 	if (
 		( defined( 'IS_WPCOM' ) && IS_WPCOM )
-		|| ( Jetpack::is_connection_ready() && Jetpack::is_module_active( 'subscriptions' ) && ! ( new Status() )->is_offline_mode() )
+		|| ( Jetpack::is_connection_ready() && ! ( new Status() )->is_offline_mode() )
```

Comment:
What are the implications of this change?

Reply:
This is something, I'm not very curtain about.
When I remove this, it runs the js code, even when the subscription module is not enabled.
However I'm not sure of the following:
- Is there any use case where the subscription is not there at all?

If yes, we need to ensure that the plugin will not load in that case, because the button to enable newsletter will not make any sense.

Also I don't know what this `register_block` does in comparison to js "register block"

Can you guide me on that?

Reply:
> Also I don't know what this `register_block` does in comparison to js "register block"
> 
> Can you guide me on that?

What exploration have you already done?

Reply:
@danielbachhuber I tried to search in field guide. 
However, I have found Gutenberg docs, and to my understanding, it seems that it registers the block (js needs to use the same name to register it as well).
Is this correct?

Reply:
@kozer What have you found in the code? And how does the user experience differ with this change applied?

Reply:
This specific part actually, lets the js client render the block.
I will add what was changed and screenshots about those changes tomorrow. 
It's just a draft and I'd like your feedback on if this is a good way to go.

Long story, short, this whole pr adds a proper UI to enable subscriptions at:
- Pre-publish panel
- Post publish panel
- Subscription block

Reply:
It seems that doing it this way will also solve https://github.com/Automattic/jetpack/issues/14621 so this may be the way to go